examples/bluetooth/ble_hid_mouse.py

```diff
@@ -0,0 +1,110 @@
+# Implements a BLE HID mouse
+
+from micropython import const
+import struct
+import bluetooth
+
+
+def ble_irq(event, data):
+    global conn_handle
+    if event == 1:
```

Comment:
```
_IRQ_CENTRAL_CONNECT = const(1)
_IRQ_CENTRAL_DISCONNECT = const(2)
```

```
if event == _IRQ_CENTRAL_CONNECT:
    print("connect")
    conn_handle, _, _ = data
elif event == _IRQ_CENTRAL_DISCONNECT:
    conn_handle = None
```

Reply:
Another suggestion would be to start again advertising upon disconnect.

```
def advertise():
    global conn_handle
    conn_handle = None
    adv = (
        b"\x02\x01\x06"
        b"\x03\x03\x12\x18"  # complete list of 16-bit service UUIDs: 0x1812
        b"\x03\x19\xc2\x03"  # appearance: mouse
        b"\x09\x09MP-mouse"  # complete local name
    )
    ble.gap_advertise(100_000, adv)
```

Create global variable conn_handle and call advertise() here (in disconnect) in addition to calling it after registering the service. Thanks for this example, it was exactly what I needed to get myself started!